assets/js/atomic/blocks/component-init.ts

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable camelcase */
+declare let __webpack_public_path__: string;
+
+/**
+ * External dependencies
+ */
+import { registerBlockComponent } from '@woocommerce/blocks-registry';
+import { lazy } from '@wordpress/element';
+import {
+	WC_BLOCKS_BUILD_URL,
+	blocksConfig,
+	isFeaturePluginBuild,
+} from '@woocommerce/block-settings';
+import { addRequireChunkTranslationsHandler } from '@woocommerce/base-utils';
+import md5 from 'md5';
+
+// Modify webpack publicPath at runtime based on location of WordPress Plugin.
+// eslint-disable-next-line prefer-const
+__webpack_public_path__ = WC_BLOCKS_BUILD_URL;
+
+const domain = isFeaturePluginBuild
+	? 'woo-gutenberg-products-block'
+	: 'woocommerce';
+
+if ( blocksConfig.locale !== 'en_US' ) {
+	addRequireChunkTranslationsHandler( {
+		domain,
+		getTranslationChunkFileUrl: ( chunkId ) => {
+			const hash = md5(
+				isFeaturePluginBuild
+					? `build/${ chunkId }.js`
+					: `packages/woocommerce-blocks/build/${ chunkId }.js`
+			);
+			return `${ blocksConfig.languageUrl }/${ domain }-${ blocksConfig.locale }-${ hash }.json`;
+		},
+		translatedChunks: [
```

Comment:
I'm not sure if having the list of translation chunks hardcoded is the best approach. For languages with incomplete translations, some of the language files for the listed chunks might be missing, which would result in 404 for the translation request.

Would it be possible to generate the translated chunks list from the back-end and provide it to the app using the `blocksConfig` object, e.g.:

```php
$translated_chunks = array_map(
	'basename',
	glob( WP_LANG_DIR . '/plugins/woo-gutenberg-products-block-' . get_locale() . '*.json' )
);
```

Reply:
We still have to do some cross-checking to know which file to load I guess right? which means we still need to generate the hash on the frontend again?  `$translated_chunks` returns 227 files for the Japanse translation of WooCommerce blocks.

Reply:
I applied this feedback @yuliyan if you want to take another look.

I'm not fully convinced of this approach. Passing the full list of files being translated adds 16kb to the page size (227) while we might need only 4 or 5. So on a later PR I might just pass down atomic blocks (the ones being dynamically loaded).

Reply:
I didn't expect the files to be that many, but yeah, in this case it doesn't make a lot of sense to provide all that data to the front-end and increase the page size unnecessarily.

As you suggested, it would be better if you could pass only the files that are relevant to the front, maybe something like that could do the job:

```
$files_with_translations = array_map(
	'md5',
	array(
		'build/atomic-block-components/add-to-cart.js',
		'build/atomic-block-components/button-frontend.js',
		'build/atomic-block-components/rating.js',
		'build/atomic-block-components/title.js',
		'build/atomic-block-components/tag-list.js',
		...
	) // hardcode the atomic block filepaths, or get them dynamically with another glob.
);
```

And then the glob pattern would be: 
`glob( WP_LANG_DIR . '/plugins/woo-gutenberg-products-block-' . get_locale() . "-{{$files_with_translations}}" . '*.json', GLOB_BRACE );`